### PR TITLE
 Resize suggestion box when the user scrolls

### PR DESCRIPTION
**Pull request by davidmartos96**
I think this is an useful feature because is not uncommon that the keyboard hides the suggestion box when the TextField is positioned in the middle of some scrollable screen and then the user tries to scroll to show the suggestions.

### CODE REVIEW

**Comment by sjmcdowall** [comment]
Thanks for the PR @davidmartos96 !  I hope to make able to give this a spin tomorrow ..

File: lib/flutter_typeahead.dart [hunk 1]
```diff
@@ -669,6 +672,7 @@ class _TypeAheadFieldState<T> extends State<TypeAheadField<T>>
   void dispose() {
     this._suggestionsBoxController.widgetMounted = false;
     WidgetsBinding.instance.removeObserver(this);
+    this._resizeOnScrollTimer?.cancel();
```
Comment (sjmcdowall) [inline lib/flutter_typeahead.dart]: Do we need this this. (for both _resize and _suggestionsBoxController) above?  I am pretty sure the Google Linting rules suggests to not use this. unless really needed... ??

File: lib/flutter_typeahead.dart [hunk 2]
```diff
@@ -705,6 +709,25 @@ class _TypeAheadFieldState<T> extends State<TypeAheadField<T>>
...
+      ScrollableState scrollableState = Scrollable.of(context);
+      if (scrollableState != null) {
+        // The TypeAheadField is inside a scrollable widget
+        scrollableState.position.isScrollingNotifier.addListener(() {
+          bool isScrolling = scrollableState.position.isScrollingNotifier.value;
+          _resizeOnScrollTimer?.cancel();
+          if (isScrolling) {
+            // Scroll started
+            _resizeOnScrollTimer =
+                Timer.periodic(Duration(milliseconds: 500), (timer) {
```
Comment (sjmcdowall) [inline lib/flutter_typeahead.dart]: Let's make the 500 ms a const variable at the top so we can easily modify it.. and .. honestly 1/2 second seems sort of long?  Maybe 300 ms or 250?  Did you try different values?

**Comment by davidmartos96** [comment]
Alright, sounds good. I added a Scrollable page in the example app so that is easy to try it.
I had to update gradle in the android folder to run the example with the latest flutter release. I didn't commit those changes though, because I didn't know if they were needed.

File: lib/flutter_typeahead.dart (hunk 1 above)
Comment (davidmartos96) [reply lib/flutter_typeahead.dart]: I don't usually write the "this" but I saw it was used on the two lines above. That's why I used the "this".

File: lib/flutter_typeahead.dart (hunk 2 above)
Comment (davidmartos96) [reply lib/flutter_typeahead.dart]: Alright. Lower values should work fine. I will add it as a const.

Comment (davidmartos96) [reply lib/flutter_typeahead.dart]: @sjmcdowall I tried lower values like 200 or 300, but those make the impression of the suggestion box lagging (On a usual scroll to show the suggestions, a low value maybe resizes the box 2 or 3 times, looking like it is drawing at low fps). A value like 500 ms leaves some room for the user to scroll and then resize only once, looking nicer.
What is your opinion?

Comment (sjmcdowall) [reply lib/flutter_typeahead.dart]: OK - lets try 500 and see how it goes..

File: lib/flutter_typeahead.dart (hunk 1 above)
Comment (davidmartos96) [reply lib/flutter_typeahead.dart]: I updated the code, removing the "this."

**Comment by davidmartos96** [comment]
@sjmcdowall I changed PR destination to develop

**Pull request merged**